### ISSUE TITLE
Crash on changing from Belts to Lanes

### ISSUE THREAD

**Issue by Nidhogg-Wyrmborn** [ClaudeMetz/FactoryPlanner#245]
### Problem Description

The mod Factory Planner (1.1.78) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::on_gui_switch_state_changed (ID 151)
__factoryplanner__/backend/calculation/solver.lua:181: attempt to index local 'factory' (a nil value)
stack traceback:
	__factoryplanner__/backend/calculation/solver.lua:181: in function 'update'
	__factoryplanner__/ui/dialogs/preferences_dialog.lua:252: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>

### Reproduction

1. open factory planner
2. click preferences
3. under "preferred belt" switch from belts to Lanes

**Comment by Hellrespawn** [comment]
I've run into this same error, even on a new save with no other mods enabled. Error message:

```
The mod Factory Planner (1.1.78) caused a non-recoverable error.
Please report this error to the mod author.

Error while running event factoryplanner::on_gui_switch_state_changed (ID 151)
__factoryplanner__/backend/calculation/solver.lua:181: attempt to index local 'factory' (a nil value)
stack traceback:
	__factoryplanner__/backend/calculation/solver.lua:181: in function 'update'
	__factoryplanner__/ui/dialogs/preferences_dialog.lua:252: in function 'handler'
	__factoryplanner__/ui/event_handler.lua:157: in function <__factoryplanner__/ui/event_handler.lua:114>
```

**Comment by KugelblitzNinja** [comment]
Factory Planner (1.1.79) - Factorio 1.1.110

Iv just had this happend to me,

Iv noticed this ONLY happens when there are no planned factory.
^ as soon as i make one this stops happning.

**Comment by ClaudeMetz** [comment]
Yep thanks everyone, this was just noticed on my 2.0 branch as well. It's fixed there, so: Fixed for 2.0!
